zipline/pipeline/data/dataset.py

```diff
@@ -250,8 +266,24 @@ def fx(self, currency):
             Column producing the same data as ``self``, but currency-converted
             into ``currency``.
         """
+        if hasattr(self.dataset, 'extra_coords'):
+            name = self.dataset.extra_coords['item']
+        else:
+            name = self.name
+
+        if self.dtype != float64_dtype:
+            raise TypeError(
+                'The .fx method cannot be called on {} because {} does not '
+                'have a float64 dtype.'.format(self, name)
+            )
+
         conversion = self._currency_conversion
-        if conversion is not None and conversion.currency == currency:
+        if conversion is Disallowed:
+            raise TypeError(
+                'The .fx method cannot be called on {} because {} is not '
```

Comment:
Thinking about the wording here:

- I'm not sure why we're including both `self` and a variant on `self.name` in this message. I think `self.qualname` should be sufficient to tell the reader what column they used.
- Do we want "is not currency-aware" to be the user-facing description of the problem here? That's a very mechanical description of the issue vs. something more conceptual like "does not produce currency-denominated data". cc @jmccorriston for thoughts on the language here.